container/src/main/java/com/flipkart/poseidon/core/PoseidonServlet.java

```diff
@@ -186,13 +187,17 @@ private void buildResponse(PoseidonRequest request, PoseidonResponse poseidonRes
         httpResponse.setStatus(statusCode);
         Object responseObj = poseidonResponse.getResponse();
         if (responseObj != null) {
-            String responseStr = "";
             if (responseObj instanceof String) {
-                responseStr = (String) responseObj;
+            	httpResponse.getWriter().println((String)responseObj);
+            } else if (responseObj instanceof ByteArrayDataType) {
+            	byte[] rawBytes = ((ByteArrayDataType)responseObj).getRawBytes();
+            	// we override default response meta-data as the data is raw bytes
+            	httpResponse.setContentType(MediaType.OCTET_STREAM.toString());
+            	httpResponse.setContentLength(rawBytes.length);
+            	httpResponse.getOutputStream().write(rawBytes);
```

Comment:
Though not inclined to adding a DataType to container (as api module actually orchestrates using lego and hydra and returns datatypes), given that in this solution, users don't need to compose at api json file to use ByteArrayDataType.rawBytes (which ends up touching multiple points to implement a file download), this looks a better approach (just wrap byes in ByteArrayDataType in DataSource and rest will be taken care).